tools/run_tests/run_interop_tests.py

```diff
@@ -843,7 +902,7 @@ def aggregate_http2_results(stdout):
     for language in languages:
       for test_case in _TEST_CASES:
         if not test_case in language.unimplemented_test_cases():
-          if not test_case in skip_server:
+          if not test_case in skip_server and not args.http2_badserver_interop:
```

Comment:
Can the check for `...and not args.http2_badserver_interop` be pushed up above the start of the for loops? This would make the flow clearer here, I think.

Also, should this also check to make sure that `args.http2_interop` is false?